d2/src/main/java/com/linkedin/d2/discovery/stores/zk/ZooKeeperStore.java

```diff
@@ -42,8 +42,9 @@
     PropertyEventPublisher<T>,
     PropertyStore<T>
 {
-  private static final Logger           _log =
-                                                 LoggerFactory.getLogger(ZooKeeperStore.class);
+  private static final Logger           _log = LoggerFactory.getLogger(ZooKeeperStore.class);
+
+  public static final int DEFAULT_READ_WINDOW_MS = -1; //disabled by default
```

Comment:
If you see the use of the read window in processWatch function - we are using this to randomize setting the watch. So we may not be able to leverage the TimeSpan here and it may introduce unwanted conversion back to int in processWatch to randomize.

Reply:
could the function "getDurationInMilliseconds" in Timespan work for getting a long in milliseconds for the calculation?